src/NLog/GlobalDiagnosticsContext.cs

```diff
@@ -57,22 +57,42 @@ public static void Set(string item, string value)
         }
 
         /// <summary>
+        /// Sets the Global Diagnostics Context item to the specified value.
+        /// </summary>
+        /// <param name="item">Item name.</param>
+        /// <param name="value">Item value.</param>
+        public static void Set(string item, object value)
+        {
+            lock (dict)
+            {
+                dict[item] = value;
+            }
+        }
+
+        /// <summary>
         /// Gets the Global Diagnostics Context named item.
         /// </summary>
         /// <param name="item">Item name.</param>
         /// <returns>The item value of string.Empty if the value is not present.</returns>
         public static string Get(string item)
         {
+            return RenderContextObject(GetObject(item));
```

Comment:
No, a `null` value will always return `String.Empty`. `String.Format` returns an empty string for "{0}" if the format argument is `null`.

I wasn't sure initially, so I tested it before writing the code this way.